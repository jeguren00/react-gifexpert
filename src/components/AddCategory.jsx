import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategory = ({onAddCategory}) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();

        if (inputValue.trim().length <= 1) {
            return;
        }
        onAddCategory(inputValue.trim());
        setInputValue('');
    }
    return (
        <form aria-label="form" onSubmit={onSubmit}>
            <input 
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={(event) => onInputChange(event)}
             />
        </form>
    )
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired,
  }