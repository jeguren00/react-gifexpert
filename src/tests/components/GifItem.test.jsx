import { fireEvent, render,screen } from "@testing-library/react"
import { GifItem } from "../../components/GifItem";

describe('pruebas con el componente gifitem', () => { 
    const title = "Imagen de prueba";
    const url = "https://upload.wikimedia.org/wikipedia/commons/7/71/1885Benz.jpg";

    test('match con el snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
     });

     test('mostrar la imagen con el url y alt indicado', () => { 
        render(<GifItem title={title} url={url}/>);

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
     });

     test('debe de mostrar el titulo en el componente', () => { 
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
      })
 })