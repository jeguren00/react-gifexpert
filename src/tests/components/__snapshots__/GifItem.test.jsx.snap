// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con el componente gifitem match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Imagen de prueba"
      src="https://upload.wikimedia.org/wikipedia/commons/7/71/1885Benz.jpg"
    />
    <p>
      Imagen de prueba
    </p>
  </div>
</div>
`;
